core/cap-0046-06.md

```diff
@@ -296,36 +226,56 @@ flow between classic and smart.
 *
 \******************************************************************************/
 // Move "amount" from "id" on classic to "id" on smart
-fn move_to_smart(id: KeyedAccountAuthorization, amount: BigInt) -> bool;
+fn import(id: Signature, nonce: BigInt, amount: i64) -> Result<(), Error>;
 
 // Move "amount" from "id" on smart to "id" on classic
-fn move_to_classic(id: KeyedAccountAuthorization, amount: BigInt) -> bool;
+fn export(id: Signature, nonce: BigInt, amount: i64) -> Result<(), Error>;
 ```
 
 ### Semantics: Deployment
 
-We introduce the following host functions to facilitate creation of these
-contracts:
+#### Deploying a contract that allow wrapping Stellar classic assets
 
-```rust
-// Creates the contract that wraps the native asset
-fn create_native_asset_wrapper();
+In order to guarantee uniqueness of wrapper contracts, `create_token_from_asset`
+produces a deterministic contract identifier that does not depend on the creator
+or any salt. This is achieved by introducing
+`ENVELOPE_TYPE_CONTRACT_ID_FROM_ASSET` in
+[CAP-0046-02](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0046-02.md#contract-identifier-preimage-types).
+The wrapper contracts should be initialized atomically with contract creation so
+the right asset is passed in. `create_token_from_asset` should call the function
```

Comment:
What function does this refer to? I would probably not refer to the functions here, this is specified currently only in XDR (also I don't think `create_token_from_asset` corresponds to anything now...)

Reply:
What I meant is that the `create_token_from_asset` doesn't exist, does it? Asset tokens currently can only be created via transactions with id from `Asset` and source set to token (i.e. it can be created only via XDR params, not a function).

Reply:
It exists in env at the moment as an exposed function, and with your deployment changes, will become internal. It's still part of the protocol though so I specified it here.

Reply:
Ah, right, this PR is yet to be sent. It won't be a part of the protocol in this sense for now (i.e. it will be a part of protocol defined in the transaction XDR only, but not in the env). I can update this later though.

Reply:
I'm curious why that distinction should be made (XDR vs env). The CAP should describe how the protocol is changing, which includes env changes (even if the change to env is not exposed to contracts).